tests/core/channels/test_twilio_voice.py

```diff
@@ -0,0 +1,313 @@
+
+
+async def test_twilio_voice_remove_image():
+
+    with pytest.warns(UserWarning):
+        output_channel = TwilioVoiceCollectingOutputChannel()
+        await output_channel.send_response(
+            recipient_id="Chuck Norris",
+            message={"image": "https://i.imgur.com/nGF1K8f.jpg", "text": "Some text."},
+        )
+
+
+async def test_twilio_voice_keep_image_text():
+
+    output_channel = TwilioVoiceCollectingOutputChannel()
+    await output_channel.send_response(
+        recipient_id="Chuck Norris",
+        message={"image": "https://i.imgur.com/nGF1K8f.jpg", "text": "Some text."},
+    )
+    assert len(output_channel.messages) == 1
+    assert output_channel.messages[0]["text"] == "Some text."
+
+
+async def test_twilio_emoji_warning():
+
+    with pytest.warns(UserWarning):
+        output_channel = TwilioVoiceCollectingOutputChannel()
+        await output_channel.send_response(
+            recipient_id="User", message={"text": "Howdy 😀"}
+        )
+
+
+async def test_twilio_voice_multiple_responses():
+
+    inputs = {
+        "initial_prompt": "hello",
+        "reprompt_fallback_phrase": "i didn't get that",
+        "speech_model": "default",
+        "speech_timeout": "5",
+        "assistant_voice": "woman",
+        "enhanced": "false",
+    }
+
+    tv = TwilioVoiceInput(**inputs)
+
+    output_channel = TwilioVoiceCollectingOutputChannel()
+
+    await output_channel.send_text_message(
+        recipient_id="Chuck Norris", text="message 1"
+    )
+    await output_channel.send_text_message(
+        recipient_id="Chuck Norris", text="message 2"
+    )
+    assert len(output_channel.messages) == 2
+    assert output_channel.messages[0]["text"] == "message 1"
+    assert output_channel.messages[1]["text"] == "message 2"
+
+    twiml = tv._build_twilio_voice_response(output_channel.messages)
+
+    assert (
+        str(twiml)
+        == '<?xml version="1.0" encoding="UTF-8"?><Response><Say voice="woman">message 1</Say><Pause length="1" /><Gather action="/webhooks/twilio_voice/webhook" actionOnEmptyResult="true" enhanced="false" input="speech" speechModel="default" speechTimeout="5"><Say voice="woman">message 2</Say></Gather></Response>'
+    )
+
+
+async def test_twilio_receive_answer(stack_agent: Agent):
+    app = server.create_app(agent=stack_agent)
+
+    inputs = {
+        "initial_prompt": "hello",
+        "reprompt_fallback_phrase": "i didn't get that",
+        "speech_model": "default",
+        "speech_timeout": "5",
+        "assistant_voice": "woman",
+        "enhanced": "false",
+    }
+
+    tv = TwilioVoiceInput(**inputs)
+    channel.register([tv], app, "/webhooks/")
+
+    client = app.asgi_client
+
+    body = {"From": "Tobias", "CallStatus": "ringing"}
+    _, response = await client.post(
+        "/webhooks/twilio_voice/webhook",
+        headers={"Content-type": "application/x-www-form-urlencoded"},
+        data=body,
+    )
+    assert response.status == 200
```

Comment:
optional (same for others)
```suggestion
    assert response.status == HTTPStatus.OK
```